File: examples/node/hello.js
/**
 * Simple Hello World example for Node.js
 */

/**
 * Return a greeting message
 * @param {string} name - The name to greet
 * @returns {string} A greeting string
 */
function greet(name = 'World') {
  return `Hello, ${name}!`;
}

/**
 * Main function
 */
function main() {
  console.log(greet());
  console.log(greet('Agent'));
}

// Export for testing
module.exports = { greet };

// Run if this is the main module
if (require.main === module) {
  main();
}

File: examples/python/hello.py
#!/usr/bin/env python3
"""Simple Hello World example for Python."""


def greet(name: str = "World") -> str:
    """Return a greeting message.
    
    Args:
        name: The name to greet. Defaults to "World".
        
    Returns:
        A greeting string.
    """
    return f"Hello, {name}!"


def main() -> None:
    """Main function."""
    print(greet())
    print(greet("Agent"))


if __name__ == "__main__":
    main()
